lib/src/acter_avatar.dart

```diff
@@ -162,245 +126,242 @@ class _ActerAvatar extends State<ActerAvatar> {
+                  onForegroundImageError: avatarError,
+                  radius: size ?? 24,
+                );
+        }));
   }
 
-  Widget renderSpaceParent(BuildContext context) {
-    if (widget.badgeSize == null) {
-      // nothing. ignore
-      return SizedBox.shrink();
-    }
-    final badgeSize = widget.badgeSize ?? 20;
-    if (widget.avatarsInfo == null || widget.avatarsInfo!.isEmpty) {
-      return SizedBox(
-        height: badgeSize,
-        width: badgeSize,
+  Widget renderBadges(
+    BuildContext context,
+    DefaultAvatarOptions options,
+    double? badgesSize,
+  ) {
+    final size = badgesSize ?? 20;
+    final badges = options.parentBadges!;
+    double badgeOverflow = size / 5;
+    final badgesLength = badges.length;
+    final thresholdCount = badgesLength <= 3 ? badgesLength : 3;
+    List<Widget> children = [];
+
+    for (int i = 0; i < thresholdCount; i++) {
+      children.add(
+        ActerAvatar(options: AvatarOptions(badges[i], size: size)),
       );
     }
-
-    final parentInfo = widget.avatarsInfo![0];
-    double badgeOverflow = badgeSize / 5;
+    if (badgesLength > 3) {
+      children.add(Align(
+        alignment: Alignment.bottomRight,
+        child: Container(
+          width: 10,
+          height: 10,
+          alignment: Alignment.center,
+          decoration: const BoxDecoration(
+            color: Colors.green,
+            shape: BoxShape.circle,
+          ),
+          child: Text(
+            '+${badgesLength - 3}',
+            style: const TextStyle(fontSize: 8),
+            textScaler: TextScaler.linear(0.7),
+          ),
+        ),
+      ));
+    }
 
     return Positioned(
       bottom: -badgeOverflow,
       right: -badgeOverflow,
       child: GestureDetector(
-        onTap: widget.onParentBadgeTap,
-        child: ActerAvatar(
-          avatarInfo: parentInfo,
-          mode: DisplayMode.Space,
-          size: badgeSize,
-          badgeSize: null,
+        onTap: widget.onParentBadgesTap,
+        child: Wrap(
+          spacing: -5,
+          children: children,
         ),
       ),
     );
   }
 
   Widget renderFallback(BuildContext context) {
-    double textFallbackSize = widget.size ?? 48;
-    double multiFallbackSize = widget.size == null ? 48 : widget.size! * 2.0;
-
-    /// Fallback
-    switch (widget.mode) {
-      case DisplayMode.DM:
-        // User fallback mode
-        return MultiAvatar(
-          uniqueId: widget.avatarInfo.uniqueId,
-          size: multiFallbackSize,
-        );
-      case DisplayMode.GroupDM:
-        return widget.avatarsInfo != null && widget.avatarsInfo!.isNotEmpty
-            ? Stack(
-                alignment: Alignment.bottomLeft,
-                clipBehavior: Clip.none,
-                children: [
-                  MultiAvatar(
-                    uniqueId: widget.avatarInfo.uniqueId,
-                    size: multiFallbackSize,
+    final avatarOptions = widget.options;
```

Comment:
The changes in methods only reflects switching avatar type fallback based on ```AvatarOptions``` type constructor instead of old way enum.